openhcl/virt_mshv_vtl/src/processor/mshv/x64.rs

```diff
@@ -704,18 +705,14 @@ impl<'a, 'b> InterceptHandler<'a, 'b> {
             == self.vp.partition.monitor_page.gpa()
             && message.header.intercept_access_type == HvInterceptAccessType::WRITE
         {
-            let instruction_bytes = message.instruction_bytes;
-            let instruction_bytes = &instruction_bytes[..message.instruction_byte_count as usize];
             let tlb_lock_held = message.memory_access_info.gva_gpa_valid()
                 || message.memory_access_info.tlb_locked();
-            let mut state = self.vp.emulator_state(self.intercepted_vtl);
-            if let Some(bit) = virt_support_x86emu::emulate::emulate_mnf_write_fast_path(
-                instruction_bytes,
-                &mut state,
-                interruption_pending,
+            if let Some(bit) = self.vp.emulate_fast_path(
                 tlb_lock_held,
```

Comment:
Is anywhere else calling emulate_mnf_write_fast_path? It'd be nice if we could keep the code here for now so we can avoid the extra instruction_byte copying and other overhead of genericizing it if nobody else needs it yet.